src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Relocation.cs

```diff
@@ -11,11 +12,14 @@ public enum RelocType
         IMAGE_REL_BASED_ABSOLUTE        = 0x00,     // No relocation required
         IMAGE_REL_BASED_ADDR32NB        = 0x02,     // The 32-bit address without an image base (RVA)
         IMAGE_REL_BASED_HIGHLOW         = 0x03,     // 32 bit address base
+        IMAGE_REL_BASED_THUMB_MOV32     = 0x07,     // Thumb2: based MOVW/MOVT
         IMAGE_REL_BASED_DIR64           = 0x0A,     // 64 bit address base
         IMAGE_REL_BASED_REL32           = 0x10,     // 32-bit relative address from byte following reloc
         IMAGE_REL_THUMB_MOV32           = 0x11,     // Thumb2: MOVW/MOVT
-        IMAGE_REL_THUMB_BRANCH24        = 0x14,     // Thumb2: B, BL   
-             
+
+        IMAGE_REL_BASED_THUMB_BRANCH24  = 0x13,     // Thumb2: based B, BL
+        IMAGE_REL_THUMB_BRANCH24        = 0x14,     // Thumb2: B, BL
```

Comment:
@hoyMS Why are we mixing `IMAGE_REL_BASED_XXX` and `IMAGE_REL_THUMB_XXX` here (it started with change 1628798 on 9/22/2016)? They are two different enums. It will results into having two different values for the same thing; and it will likely result into collisions pretty soon as well.

@sergign60 Could you please keep `IMAGE_REL_THUMB_XXX` together, and add TODO with a link to a issue to get it cleaned up?